js/compiler.ts

```diff
@@ -383,21 +447,29 @@ class Host implements ts.CompilerHost {
     onError?: (message: string) => void,
     shouldCreateNewSourceFile?: boolean
   ): ts.SourceFile | undefined {
-    assert(!shouldCreateNewSourceFile);
-    util.log("getSourceFile", fileName);
-    const sourceFile =
-      fileName in this._sourceFileCache
-        ? this._sourceFileCache[fileName]
-        : this._resolveModule(fileName, ".");
-    assert(sourceFile != null);
-    if (!sourceFile.sourceCode) {
+    util.log("compiler::host.getSourceFile", fileName);
+    try {
+      assert(!shouldCreateNewSourceFile);
+      const sourceFile = fileName.startsWith(ASSETS)
+        ? this._getAsset(fileName)
+        : SourceFile.get(fileName);
+      assert(sourceFile != null);
+      return (
+        sourceFile!.tsSourceFile ||
+        (sourceFile!.tsSourceFile = ts.createSourceFile(
```

Comment:
Can you move this bit before `return`? It obscures the view of what's going on here